aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java

```diff
@@ -98,7 +99,7 @@ public AwsClientProperties() {
   }
 
   public AwsClientProperties(Map<String, String> properties) {
-    this.allProperties = SerializableMap.copyOf(properties);
+    this.allProperties = new HashMap<>(properties);
```

Comment:
This would serializableMap itself is not kryo serializable ? hmm which can break other things ? 

how about adding a test to this meanwhile there are KryoSerde Test in S3FileIO

Reply:
in fact we're using `SerializableMap` mainly due to Kryo ser/de so I'm surprised that this would fix the issue. Can you please add a test to `TestS3FileIO` that reproduces the issue you're running into?